Framework/Core/src/TableTreeHelpers.cxx

```diff
@@ -454,7 +454,7 @@ std::shared_ptr<TTree> TableToTree::process()
   }
 
   for (auto& reader : mColumnReaders) {
-    int idealBasketSize = 1024 + reader->fieldSize() * mRows; // minimal additional size needed, otherwise we get 2 baskets
+    int idealBasketSize = 1024 + reader->fieldSize() * reader->columnEntries(); // minimal additional size needed, otherwise we get 2 baskets
```

Comment:
Probably it would be better to set field size for the column correctly in the constructor. Currently it is a bare byte width of the element type, in case of a fixed list it is multiplied by the list size. For the variable-size list we can use `columnEntries / mRows` as an estimate of an average list size.